internal/engine/buildcontroller.go

```diff
@@ -92,6 +95,19 @@ func nextTargetToBuild(state store.EngineState) *store.ManifestTarget {
 	return choice
 }
 
+func findUnresourcedYAMLAndPutItFirst(input []*store.ManifestTarget) {
+	for i := range input {
+		if input[i].Manifest.ManifestName() == model.UnresourcedYAMLManifestName {
+			found := input[i]
+			// Remove found
+			input = append(input[:i], input[i+1:]...)
```

Comment:
I thiiiink:
1. it's not generally safe to assign to a slice param and expect the changes to affect the caller
2. it happens to work most or all of the time in this case due to implementation details

Maybe I'm just not sufficiently comfortable with slice internals, but I'd prefer to either return the modified slice, or limit operations in this function to strictly reordering elements rather than reassigning to `input`, e.g. (untested):
```
found := input[i]
copy(input[1:], append(input[:i], input[i+1:]...))
input[0] = found
```

Reply:
That works for me!